### PR TITLE
Install and Setup Devise for User Authentication

### PR DESCRIPTION
**Pull request by peter-akhigbe**
- Can register a new user. ✅
- User logs in with a combination of email and password. ✅
- Hashed password should be stored in the database. ✅
- Ask for confirmation of email. ✅
- Can reset password. ✅

- Remove current_user method in ApplicationController. ✅

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @peter-akhigbe, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have passing linters, kindly fix it.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Reem-lab** [comment]
Hello @peter-akhigbe could you check your slack?

**Comment by peter-akhigbe** [comment]
Hello @ reviewer

I did the project solo because of some issues with my coding partner.

Below is a screenshot of the email from Marco Lizza about the issue.

So please kindly disregard the collaboration requirements for this project

Thank you.

image

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,15 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (Reem-lab) [inline app/models/user.rb:5]: - [ ] Great job for adding the confirmation after signing up I can see the user index page directly but it should lead me to a confirmation mail so after the confirmation I can log in 🎉🔥🚀


image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @peter-akhigbe, 👋🏻👋🏻 

Great job so far! 🔥✨💯⭐
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No Linters error 🚀
- [ ] Followed GitFlow 💯
- [ ] Amazing job working on setup the devise👏🏻👏🏻




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @peter-akhigbe, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
Fantastic! 🎉 Here's a revised version of the code review feedback:

1. **Confirmation Implementation:**
   - Kudos on swiftly implementing the email confirmation step after signing up! 🚀 This will greatly enhance the security and reliability of user accounts.
   - Verify that users are now redirected to a confirmation mail before attempting to log in.

2. **Authentication Adjustment:**
   - Well done on adding the `authenticate_user!` method to the `ApplicationController`. This ensures that users need to confirm their email before gaining access, boosting the overall security of your application.

3. **User Experience Enhancement:**
   - The changes made contribute to a smoother and more secure user experience, aligning with best practices in user authentication and verification.

4. **Code Quality:**
   - Confirm that any relevant documentation has been updated to reflect the recent changes. Documentation is crucial for maintaining codebase clarity.
   - Double-check the implementation to ensure a seamless and user-friendly confirmation process.

5. **Celebration:**
   - 🎉 Outstanding work! Your responsiveness and dedication to improving the application are commendable. Keep up the excellent effort!
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**